spinalcordtoolbox/image.py

```diff
@@ -1854,28 +1854,30 @@ def check_image_kind(img):
 
         - 'seg': Binary segmentation (0/1)
         - 'softseg': Nonbinary segmentation in the range [0, 1], where 0 and 1 are the majority of values
-        - 'labeled_seg': Nonbinary, whole values (0, 1, 2...) or (0.0, 1.0, 2.0...) only
-        - 'im': Any other float-valued image where 0 and 1 are not the majority of values.
+        - 'labeled_seg': Nonbinary, whole values (0, 1, 2...) where 0 is the majority of values
+        - 'im': Any other image where 0 is not the majority of values.
 
     Useful for determining A) which colormap should be applied to an image, or
                            B) which interpolation can be safely used on a given image
 
     Output strings should be compatible with display_viewer_syntax() function.
     """
-    # Check if image is a segmentation (binary or soft) by making sure:
-    # - 0/1 are the two most common voxel values
-    # - 0/1 account for >95% of voxels (to allow for some soft voxels)
+    # Count the unique values in the image
     unique, counts = np.unique(np.round(img.data, decimals=1), return_counts=True)
     unique, counts = unique[np.argsort(counts)[::-1]], counts[np.argsort(counts)[::-1]]  # Sort by counts
+    # This heuristic helps to detect binary and soft segmentations
     binary_most_common = set(unique[0:2].astype(float)) == {0.0, 1.0}
     binary_percentage = np.sum(counts[0:2]) / np.sum(counts)
+    # This heuristic helps to distinguish between PSIR images and label images (2-10% zero vs. 99% zero)
     is_whole_only = np.equal(np.mod(unique, 1), 0).all()
+    zero_most_common = float(unique[0]) == 0.0
+    zero_percentage = np.sum(counts[0]) / np.sum(counts)
     if binary_most_common:
         if binary_percentage == 1.0:
             return 'seg'
         elif binary_percentage > 0.95:
             return 'softseg'
-    if is_whole_only:
+    elif is_whole_only and zero_most_common and zero_percentage > 0.50:
```

Comment:
The more I think about it, the more I'm convinced that this should be an `if`, rather than `elif`. The heuristics we're using have complicated logical relationships, and we can't hope for a neat `if/elif/else` separation. It's more like a decision tree.

Also, `zero_percentage > 0.50` should imply `zero_most_common == True`, so we don't have to compute `zero_most_common` at all.

Reply:
Sorry about the unnecessary change! I had looked at the second `if` and thought "huh, this is strange, why had I originally written it this way?" and changed it. But, it looks like you've answered the question for me.